backend/users/views.py

```diff
@@ -100,41 +165,46 @@ def get(self, request, *args, **kwargs):
             client_id, token={"access_token": access_token}
         ).get(user_url)
         user_data = user_response.json()
-
         user = {
             "id": user_data["id"],
             "login": user_data["login"],
-            # "avatar": user_data["image"]["versions"]["small"],
+            "email": user_data["email"],
             "access_token": access_token,
         }
 
         avatar_url = user_data["image"]["versions"]["small"]
         response = requests.get(avatar_url)
 
         User = get_user_model()
-        username = f"{user['login']}#{user['id']}"
+        username = f"{user['login']}{user['id']}"
 
         try:
-            existing_user = User.objects.get(username=username)
+            existing_user = User.objects.get(username=username, email=user["email"])
         except User.DoesNotExist:
             # If the user does not exist, create a new user
-            existing_user = User.objects.create(username=username)
+            existing_user = User.objects.create(username=username, email=user["email"])
         else:
             existing_user.avatar.delete()  # Delete the old avatar
 
         # Save the new avatar
         existing_user.avatar.save(f"{username}.jpg", ContentFile(response.content))
```

Comment:
Is it safe if someone changes his username and somebody steal the username and change the image ?